build-logic/performance-testing/src/main/groovy/gradlebuild/performance/generator/MavenJarCreator.groovy

```diff
@@ -53,24 +54,32 @@ class MavenJarCreator {
         }
     }
 
-    private void addJarEntry(JarOutputStream out, String name, String content) {
+    private void addZipEntry(ZipOutputStream out, String name, String content) {
         // Add archive entry
-        JarEntry entry = new JarEntry(name)
-        entry.setTime(System.currentTimeMillis())
+        ZipEntry entry = new NormalizedZipEntry(name)
         out.putNextEntry(entry)
 
         // Write file to archive
         def contentBytes = content.getBytes("utf-8")
         out.write(contentBytes, 0, contentBytes.length)
     }
 
-    private void addGeneratedUncompressedJarEntry(JarOutputStream out, String name, int sizeInBytes) {
-        JarEntry entry = new JarEntry(name)
-        entry.setTime(System.currentTimeMillis())
+    private void addGeneratedUncompressedZipEntry(ZipOutputStream out, String name, int sizeInBytes) {
+        ZipEntry entry = new NormalizedZipEntry(name)
         out.putNextEntry(entry)
 
         for (int i = 0; i < sizeInBytes; i++) {
             out.write(charsToUse, i % charsToUse.length, 1)
         }
     }
+
+    class NormalizedZipEntry extends ZipEntry {
```

Comment:
I suppose this class can be `private`. I think I would prefer a factory method instead of a subclass. I suppose you are using a subclass here since the `setTime()` methods behave strange for `0`?

Reply:
good points 

`setTime(0)` isn't enough - setting those three attributes is the minimal set of calls (I experimented with this before) that normalizes the ZIP entry.